generator_process/actions/outpaint.py

```diff
@@ -0,0 +1,74 @@
+from typing import Tuple, Generator
+from numpy.typing import NDArray
+import numpy as np
+from .prompt_to_image import ImageGenerationResult
+
+def outpaint(
+    self,
+
+    image: NDArray | str,
+
+    width: int,
+    height: int,
+
+    outpaint_origin: Tuple[int, int],
+
+    **kwargs
+) -> Generator[ImageGenerationResult, None, None]:
+    from PIL import Image, ImageOps
+
+    init_image = Image.open(image) if isinstance(image, str) else Image.fromarray(image)
+    
+    if outpaint_origin[0] > init_image.size[0] or outpaint_origin[0] < -width:
+        raise ValueError(f"Outpaint origin X ({outpaint_origin[0]}) must be between {-width} and {init_image.size[0]}")
+    if outpaint_origin[1] > init_image.size[1] or outpaint_origin[1] < -height:
+        raise ValueError(f"Outpaint origin Y ({outpaint_origin[1]}) must be between {-height} and {init_image.size[1]}")
+    
+    outpaint_bounds = Image.new(
+        'RGBA',
+        (
+            max(init_image.size[0], outpaint_origin[0] + width),
+            max(init_image.size[1], outpaint_origin[1] + height),
```

Comment:
Image gets cropped when outpainting left or up.
```suggestion
            max(init_image.size[0], outpaint_origin[0] + width) - min(0, outpaint_origin[0]),
            max(init_image.size[1], outpaint_origin[1] + height) - min(0, outpaint_origin[1]),
```